paddle/fluid/platform/device_context.cc

```diff
@@ -849,6 +849,15 @@ unsigned int MKLDNNDeviceContext::GetCachedObjectsNumber(void) const {
   return num_entries;
 }
 
+// TODO(jczaja): Replace with C++20 equivalents when applicable
+#ifdef _WIN32
+#define likely(expr) (expr)
+#define unlikely(expr) (expr)
+#else
+#define likely(expr) (__builtin_expect(!!(expr), 1))
```

Comment:
Why are using double !! ??

Reply:
__builtin_expect does accept long integer arguments. Comparison among boolean and int may raise a warning. So it was "double negated" so True is converted to 1, False to 0 and compared with long in of 1 or 0.